script.js

```diff
@@ -59,4 +59,17 @@ function switchSplitView(){
     }
 };
 
+jQuery(function() {
+    var $chkBox = jQuery('input[name="use_custom_id"]');
+    var $idRow = jQuery('input[name="target_id"]').parent();
+    if (!$chkBox.length) {
+        return;
+    }
+
+    $idRow.hide();
+    $chkBox.change(function() {
+        $idRow.toggle();
+    });
```

Comment:
This is a nice solution.

There might be a problem if you check the checkbox, then the back button in the browser, then the forward button. Normally the browser rembembers that the checkbox is checked. Now if the idRow is hidden and the checkbox is checked we have a strange situation. This might not be a problem but you can test it if you want.